### DISCUSSION TITLE
Is there a correct way to wrap multiple inputs with a single reference (React)?

### DISCUSSION THREAD

**Discussion by AlexKDawson** [floating-ui/floating-ui#2573] [Q&A] [answered]
Hey all,

So I'm trying to create a component where two inputs need to share a single drop down and the focus is trapped between the two elements and the contents of the dropdown.

I have an example that mostly gets there by wrapping the two inputs in a div that acts as the reference element. (Surprisingly the useHook interaction more or less works here). The issue is when I try to do focus trapping with FloatingFocusManager, the manager traps the focus to just the reference element.

Is there a way I can achieve what I'm attempting here, or should I just ditch it and go for a different design?

https://codesandbox.io/s/floating-ui-react-dom-template-forked-xrm8dg?file=/src/App.js

**Comment by atomiks** [comment, marked as answer]
Yeah, with the current library's code this won't work. It expects the reference element itself to be focusable and ignores any focusable children. The focus hook works due to React's event bubbling here.

Possibly an edge case we should try to handle.

**Reply by AlexKDawson** [reply]
Figured as much. Thanks for confirming!
For some context, if it was something you'd wanna consider supporting: my use case is making a date range picker, with two inputs sharing an calendar UI.

Being able to support :focus-within as opposed to just focus would be nice, but I can definitely redesign to work around this 😁

**Reply by atomiks** [reply]
I'd consider this a bug, as it should try to select focusable children if the reference itself is not focusable (especially given the focus hook uses bubbling to work). Can file an issue to track this.
